dango/account/margin/src/execute.rs

```diff
@@ -47,3 +66,166 @@ pub fn receive(_ctx: MutableCtx) -> StdResult<Response> {
     // Do nothing, accept all transfers.
     Ok(Response::new())
 }
+
+pub fn liquidate(ctx: MutableCtx, liquidation_denom: Denom) -> anyhow::Result<Response> {
+    let app_cfg: AppConfig = ctx.querier.query_app_config()?;
+
+    // Query account health
+    let HealthResponse {
+        total_debt_value,
+        utilization_rate,
+        total_collateral_value: _,
+        total_adjusted_collateral_value,
+        debts,
+        collaterals,
+    } = ctx
+        .querier
+        .query_health(ctx.contract, Some(ctx.funds.clone()))?;
+
+    // Ensure account is undercollateralized
+    ensure!(
+        utilization_rate > Udec128::ONE,
+        "account is not undercollateralized"
+    );
+
+    let health_factor = utilization_rate.checked_inv()?;
+    let target_health_factor = app_cfg.target_utilization_rate.checked_inv()?;
+    let liquidation_collateral_power = *app_cfg
+        .collateral_powers
+        .get(&liquidation_denom)
+        .ok_or_else(|| anyhow!("collateral power not found for chosen collateral"))?
+        .inner();
+
+    // Calculate liquidation bonus.
+    let bonus_cap = (total_adjusted_collateral_value
+        .checked_div(total_debt_value.checked_mul(liquidation_collateral_power)?))?
+    .saturating_sub(Udec128::ONE);
```

Comment:
Formatter is not working here? The line with `.saturating_sub` have the wrong indentation

Reply:
Seems like there's an extra set of parenthesis